core/src/main/java/io/fabric8/maven/core/config/ProcessorConfig.java

```diff
@@ -121,24 +83,117 @@ public boolean use(String name) {
      * @throws IllegalArgumentException if the includes reference an non existing element
      */
-    public <T extends Named> List<T> order(List<T> namedList, String type) {
-        if (includes == null) {
-            return namedList;
-        }
+    public <T extends Named> List<T> prepareProcessors(List<T> namedList, String type) {
         List<T> ret = new ArrayList<>();
         Map<String, T> lookup = new HashMap<>();
         for (T named : namedList) {
             lookup.put(named.getName(), named);
         }
         for (String inc : includes) {
-            T named = lookup.get(inc);
-            if (named == null) {
-                throw new IllegalArgumentException("No " + type + " with name '" + inc +
-                                                   "' found to include. " +
-                                                   "Please check spelling and your project dependencies");
+            if (use(inc)) {
+                T named = lookup.get(inc);
+                if (named == null) {
+                    List<String> keys = new ArrayList<>(lookup.keySet());
+                    Collections.sort(keys);
+                    throw new IllegalArgumentException(
+                        "No " + type + " with name '" + inc +
+                        "' found to include. " +
+                        "Please check spelling in your profile / config and your project dependencies. Included " + type + "s: " +
+                        StringUtils.join(keys,", "));
+                }
+                ret.add(named);
+            }
+        }
+        return ret;
+    }
+
+    public boolean use(String inc) {
+        return !excludes.contains(inc) && includes.contains(inc);
+    }
+
+    /**
+     * Merge in another processor configuration, with an higher priority. This operation mutates
+     * the current config.
+     *
+     * @param config config to merge in
+     * @return this merged configuration for convenience of chaining and returning.
+     */
+    public static ProcessorConfig mergeProcessorConfigs(ProcessorConfig ... processorConfigs) {
+        // Merge the configuration
+        Map<String, TreeMap> configs = mergeConfig(processorConfigs);
+
+        // Get all includes
+        Set<String> excludes = mergeExcludes(processorConfigs);
+
+        // Find the set of includes, which are the ones from the profile + the ones configured
+        List<String> includes = mergeIncludes(processorConfigs);
+
+        return new ProcessorConfig(includes, excludes, configs);
+    }
+
+    private static Set<String> mergeExcludes(ProcessorConfig ... configs) {
+        Set<String> ret = new HashSet<>();
+        for (ProcessorConfig config : configs) {
+            if (config != null) {
+                Set<String> excludes = config.excludes;
+                if (excludes != null) {
+                    ret.addAll(excludes);
+                }
             }
-            ret.add(named);
         }
         return ret;
```

Comment:
No, because its a plain Set (so simply overwriting an existing element). The set is only used to determine whether an enricher should be excluded, no order is needed here.